nilearn/tests/test_masking.py

```diff
@@ -482,6 +482,99 @@ def test_intersect_masks():
     mask_abc_ = intersect_masks([mask_a_img, mask_b_img, mask_c_img])
     assert_array_equal(mask_abc, get_data(mask_abc_))
 
+def test_label_masker():
+    ''' test the NiftiLabelsMasker module
+    '''
+    # Create a dummy atlas with 3 regions
+    atlas = np.zeros((4, 4, 2), dtype=int)
+    atlas[2:4, 2:4, :] = 1
+    atlas[3, 0:2, :] = 2
+    atlas[0:2, 3, :] = 2
+    atlas[1, 1, :] = 3
+    atlas[2, 1, :] = 3
+    atlas[1, 2, :] = 3
+    atlas_img = Nifti1Image(atlas.astype("int32"), np.eye(4))
+
+    labels = ['region1', 'region2', 'region3', 'region4']
+
+    # the atlas has two layers 
+    # of the following pattern
+    # +---+---+---+---+
+    # | 0 | 0 | 0 | 2 |
+    # +---+---+---+---+
+    # | 0 | 3 | 3 | 2 |
+    # +---+---+---+---+
+    # | 0 | 3 | 1 | 1 |
+    # +---+---+---+---+
+    # | 2 | 2 | 1 | 1 |
+    # +---+---+---+---+
+
+    # Create a dummy functional image
+    X = np.zeros((4, 4, 2, 1), dtype=int)
+    X[2:4, 2:4, :, :] = 1
+    X[3, 0:2, :, :] = 2
+    X[0:2, 3, :, :] = 2
+    X[1, 1, :, :] = 3
+    X[2, 1, :, :] = 3
+    X[1, 2, :, :] = 3
+    X = np.tile(X, (1, 1, 1, 10))
+    img = Nifti1Image(X.astype("int32"), np.eye(4))
+
+    # the img has the same 3D 
+    # as the atlas but repeated in 
+    # 10 time points, i.e., region i
+    # voxels always have value i in 
+    # img
+    # +---+---+---+---+
+    # | 0 | 0 | 0 | 2 |
+    # +---+---+---+---+
+    # | 0 | 3 | 3 | 2 |
+    # +---+---+---+---+
+    # | 0 | 3 | 1 | 1 |
+    # +---+---+---+---+
+    # | 2 | 2 | 1 | 1 |
+    # +---+---+---+---+
+
+    # apply atlas_img on img
+    masker = NiftiLabelsMasker(atlas_img,
+                           labels=labels,
+                           standardize=False
+                           )
+    masker.fit()
+    signals = masker.transform(img)
+    expected_signal = np.tile(np.array([[1, 2, 3]]), (10, 1))
+    assert_array_equal(expected_signal, signals)
```

Comment:
```suggestion
    assert_array_equal(signals, expected_signal)
```

I have seen both in the nilearn test suite, but I generally find

```python
assert results == expected_result
```

more readable than 

```python
assert expected_result == results
```

For the same reason that we more often write

```python
if length >= 10:
```

than this

```python
if 10 <= length:
```

---

The rule from: ["the art of readable code"](https://mcusoft.files.wordpress.com/2015/04/the-art-of-readable-code.pdf)

### Left-hand side	
The expression “being interrogated,” whose value is more in flux.

### Right-hand side
The expression being compared against, whose value is more constant.

Reply:
Thanks @Remi-Gau  for explaining this !